Spigot-Server-Patches/0696-Add-EntityExtinguishEvent.patch

```diff
@@ -0,0 +1,210 @@
+ 
+         if (this.inGround && !flag) {
+diff --git a/src/main/java/net/minecraft/world/level/block/BlockCauldron.java b/src/main/java/net/minecraft/world/level/block/BlockCauldron.java
+index 66d0a08f250fc947ec2d978a527ad77e32114b84..566930b55835d06bfcc0403961c244fbc6503bdf 100644
+--- a/src/main/java/net/minecraft/world/level/block/BlockCauldron.java
++++ b/src/main/java/net/minecraft/world/level/block/BlockCauldron.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.world.level.block;
+ 
++import io.papermc.paper.event.entity.EntityExtinguishEvent;
+ import net.minecraft.core.BlockPosition;
+ import net.minecraft.server.level.EntityPlayer;
+ import net.minecraft.sounds.SoundCategory;
+@@ -34,6 +35,7 @@ import net.minecraft.world.phys.shapes.VoxelShape;
+ import net.minecraft.world.phys.shapes.VoxelShapeCollision;
+ import net.minecraft.world.phys.shapes.VoxelShapes;
+ 
++import org.bukkit.craftbukkit.event.CraftEventFactory;
+ import org.bukkit.event.block.CauldronLevelChangeEvent; // CraftBukkit
+ 
+ public class BlockCauldron extends Block {
+@@ -67,7 +69,12 @@ public class BlockCauldron extends Block {
+             if (!this.changeLevel(world, blockposition, iblockdata, i - 1, entity, CauldronLevelChangeEvent.ChangeReason.EXTINGUISH)) {
+                 return;
+             }
++            // Paper start - add EntityExtinguishEvent
++            EntityExtinguishEvent event = CraftEventFactory.callEntityExtinguishEvent(entity, EntityExtinguishEvent.Cause.CAULDRON);
++            if (!event.isCancelled()) {
+             entity.extinguish();
++            }
++            // Paper end
+             // this.a(world, blockposition, iblockdata, i - 1);
+             // CraftBukkit end
+         }
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 926440e846eff2c1aaa262aa2b3975b7dd225332..95363865916ec58a9403c7f67b2b06b921faf762 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -13,6 +13,8 @@ import java.util.List;
+ import java.util.Map;
+ import java.util.stream.Collectors;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.event.entity.EntityExtinguishEvent;
+ import net.minecraft.core.BlockPosition;
+ import net.minecraft.core.EnumDirection;
+ import net.minecraft.network.protocol.game.PacketPlayInCloseWindow;
+@@ -1432,6 +1434,13 @@ public class CraftEventFactory {
+         world.getServer().getPluginManager().callEvent(event);
+         return event;
+     }
++    // Paper start
++    public static EntityExtinguishEvent callEntityExtinguishEvent(Entity entity, EntityExtinguishEvent.Cause cause) {
++        EntityExtinguishEvent event = new EntityExtinguishEvent(entity.getBukkitEntity(), cause);
++        entity.world.getServer().getPluginManager().callEvent(event);
++        return event;
++    }
++    // Paper end
```

Comment:
Nitpick: you can replace all cases of
```
EntityExtinguishEvent event = CraftEventFactory.callEntityExtinguishEvent(entity, ...);
if (!event.isCancelled()) {
```
with
```
if (new EntityExtinguishEvent(entity.getBukkitEntity(), ...).callEvent()) {
```
That way you don't pollute the local scope with an `event` variable and your patch modifies one less file.